libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt

```diff
@@ -111,3 +125,42 @@ fun createNavigationRoutes(
 fun RouteInterface.toNavigationRoute(): NavigationRoute {
     return this.toNavigationRoute()
 }
+
+private fun List<LegStep>.updateSteps(
+    route: DirectionsRoute,
+    mergedAnnotation: LegAnnotation?
+): List<LegStep> {
+    val mergedDurations = mergedAnnotation?.duration() ?: return this
+    val result = mutableListOf<LegStep>()
+    var previousStepsAnnotationsCount = 0
+    for (step in this) {
+        val stepPointsSize = route.stepGeometryToPoints(step).size
+        if (stepPointsSize < 2) {
+            logE(
+                "step at index has less than 2 points, unable to update duration",
```

Comment:
Missing the index information. We can follow up separately to avoid CI cycles though.

Reply:
Thanks. Fixed it as I had to rebase on main because of conflict in changelog